pyaedt/edb_core/nets.py

```diff
@@ -356,24 +334,50 @@ def plot(
                         if color_index >= len(CSS4_COLORS):
                             color_index = 0
                         # create patch from path
-                        patch = PathPatch(path, color=label_colors[label], alpha=0.4, label=label)
+                        objects_lists.append([vertices, codes, label_colors[label], label, 0.4, "path"])
                     else:
                         # create patch from path
-                        patch = PathPatch(path, color=label_colors[label], alpha=0.4)
+                        objects_lists.append([vertices, codes, label_colors[label], "", 0.4, "path"])
+        end_time = time.time() - start_time
+        self._logger.info("Nets Point Generation time %s seconds", round(end_time, 3))
+        return objects_lists
 
-                # plot the patch
-                ax.add_patch(patch)
+    @aedt_exception_handler
+    def plot(
+        self, nets, layers=None, color_by_net=False, show_legend=True, save_plot=None, outline=None, size=(2000, 1000)
+    ):
+        """Plot a Net to Matplotlib 2D Chart.
 
-        ax.set(xlabel="X (m)", ylabel="Y (m)", title=self._pedb.active_cell.GetName())
-        if show_legend:
-            ax.legend()
-        ax.axis("equal")
-        end_time = time.time() - start_time
-        self._logger.info("Plot Generation time %s seconds", round(end_time, 3))
-        if save_plot:
-            plt.savefig(save_plot)
-        else:
-            plt.show()
+        Parameters
+        ----------
+        nets : str, list
+            Name of the net or list of nets to plot. If `None` all nets will be plotted.
```

Comment:
@maxcapodi78 Maybe the default value should be a string: "all" instead of None to indicate that by default all layers or nets are plotted.

Reply:
This is a good idea but maybe we should think to a biggest refactoring since this approach (with None) is widely used in pyaedt and users expect the same behaviour.